flask_oidc/__init__.py

```diff
@@ -187,11 +196,115 @@ def init_app(self, app):
         except KeyError:
             pass
 
-    def load_secrets(self, app):
-        # Load client_secrets.json to pre-initialize some configuration
-        return _json_loads(open(app.config['OIDC_CLIENT_SECRETS'],
-                                   'r').read())
-        
+    def init_provider(self, provider):
+        """
+        Do setup for a specific provider
+
+        :param provider: The provider to initialize.
+        :type provider: Dictionary with at lease 'base_url' item
+        """
+
```

Comment:
Maybe we want to add a check what there's no secrets loaded yet?
Since otherwise we might overwrite existing config without people being aware.